File: java/203_Remove_Linked_List_Elements.java
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode dummy = new ListNode();
        ListNode prev = dummy, curr = head;
        prev.next = curr;
        
        while(curr!=null){
            if(curr.val == val){
                prev.next = curr.next;
                curr = curr.next;
            }
            else{
                prev = curr;
                curr = curr.next;
            }
        }
        return dummy.next;
        
    }
}

File: python/203_Remove_Linked_List_Elements.py
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        dummy = ListNode()
        prev = dummy
        prev.next, ptr = head, head
        while ptr:
            if ptr.val == val:
                ptr = ptr.next
                prev.next = ptr
            else:
                prev, ptr = ptr, ptr.next
        return dummy.next
